src/ui/include/EditorNS/editor.h

```diff
@@ -317,47 +550,72 @@ namespace EditorNS
         void getSelectedTexts(std::function<void(const QStringList&)> callback);
 
         /**
-         * @brief Get the current editor language.
-         * @param callback Lambda or functor.
-         *                 Must accept QString type as parameter.
-         */
-        void getLanguage(std::function<void(const QString&)> callback); 
-        /**
          * @brief Get the currently selected text, or the word under the cursor
          *        if no text is selected.
          * @param callback Lambda or functor.
          *                 Must accept QStringList type as parameter.
          */
+
         void getCurrentWordOrSelections(std::function<void(const QStringList&)> callback); 
 
+        /**
+         * @brief Get the current content of the editor.
+         * @param callback Lambda or functor.
+         *        Must accept QString as type parameter.
+         */
+        void getValue(std::function<void(const QString&)> callback);
+
+        /**
+         * @brief Initialize the language cache for fast access.
+         */
+        static void initLanguageCache();
+
+
+        /**
+         * @brief  Workaround. Using isClean() right after creating an Editor always returns true.
+         *         This value gets set to 'true' if the Editor starts out as a newly created empty
+         *         document, and is 'false' if the document was loaded from any file.
+         * @return
+         */
+        bool isInitiallyBlank() const { return m_initiallyBlank; }
+        void setInitiallyBlank(bool blank) { m_initiallyBlank = blank; }
```

Comment:
@JuBan1, the great thing about the new design I have planned for that includes extremely useful things, such as being able to queue page loading to the background, and emitting a signal telling it to create a tab when it's done (This should give sessions much better integration). 

Another useful thing we could do to speed up the initial load (another PR in itself), would be to thread DocEngine once we get all the GUI cruft out of it, which would stop it from blocking the UI load during session initialization.